tests/Routing/RoutingUrlGeneratorTest.php

```diff
@@ -867,6 +867,41 @@ public function testRouteGenerationWithBackedEnums()
 
         $this->assertSame('http://www.foo.com/foo/fruits', $url->route('foo.bar', CategoryBackedEnum::Fruits));
     }
+
+    public function testSignedUrlWithKeyResolver()
+    {
+        $url = new UrlGenerator(
+            $routes = new RouteCollection,
+            $request = Request::create('http://www.foo.com/')
+        );
+        $url->setKeyResolver(function () {
+            return 'secret';
+        });
+
+        $route = new Route(['GET'], 'foo', ['as' => 'foo', function () {
+            //
+        }]);
+        $routes->add($route);
+
+        $request = Request::create($url->signedRoute('foo'));
+
+        $this->assertTrue($url->hasValidSignature($request));
+
+        $request = Request::create($url->signedRoute('foo').'?tempered=true');
```

Comment:
I think there is a typo, it should probably say `tampered`

Reply:
Can you send in a PR?

Reply:
PR for fixing the typo: https://github.com/laravel/framework/pull/51238